src/java.base/share/classes/java/lang/foreign/GroupLayout.java

```diff
@@ -45,55 +40,7 @@
  * @since 19
  */
 @PreviewFeature(feature=PreviewFeature.Feature.FOREIGN)
-public final class GroupLayout extends AbstractLayout implements MemoryLayout {
-
-    /**
-     * The group kind.
-     */
-    enum Kind {
-        /**
-         * A 'struct' kind.
-         */
-        STRUCT("", Math::addExact),
-        /**
-         * A 'union' kind.
-         */
-        UNION("|", Math::max);
-
-        final String delimTag;
-        final LongBinaryOperator sizeOp;
-
-        Kind(String delimTag, LongBinaryOperator sizeOp) {
-            this.delimTag = delimTag;
-            this.sizeOp = sizeOp;
-        }
-
-        long sizeof(List<MemoryLayout> elems) {
-            long size = 0;
-            for (MemoryLayout elem : elems) {
-                size = sizeOp.applyAsLong(size, elem.bitSize());
-            }
-            return size;
-        }
-
-        long alignof(List<MemoryLayout> elems) {
-            return elems.stream().mapToLong(MemoryLayout::bitAlignment).max() // max alignment in case we have member layouts
-                    .orElse(1); // or minimal alignment if no member layout is given
-        }
-    }
-
-    private final Kind kind;
-    private final List<MemoryLayout> elements;
-
-    GroupLayout(Kind kind, List<MemoryLayout> elements) {
-        this(kind, elements, kind.alignof(elements), Optional.empty());
-    }
-
-    GroupLayout(Kind kind, List<MemoryLayout> elements, long alignment, Optional<String> name) {
-        super(kind.sizeof(elements), alignment, name);
-        this.kind = kind;
-        this.elements = elements;
-    }
+public sealed interface GroupLayout extends MemoryLayout permits StructLayout, UnionLayout {
 
     /**
      * Returns the member layouts associated with this group.
```

Comment:
We may use {@return the member layouts associated with this group}.

Reply:
Same - I'll leave these tweaks for later.